contrib/mesos/pkg/scheduler/podtask/procurement.go

```diff
@@ -50,77 +62,212 @@ type AllOrNothingProcurement []Procurement
 
 // Procure runs each Procurement in the receiver list. The first Procurement func that
 // fails triggers T.Reset() and the error is returned, otherwise returns nil.
-func (a AllOrNothingProcurement) Procure(t *T, offer *mesos.Offer) error {
+func (a AllOrNothingProcurement) Procure(t *T, offer *mesos.Offer, n *api.Node, spec *Spec) error {
```

Comment:
T.Spec is the result of a successful procurement which is persistent into the podtask by the scheduler. The spec in the Procure function instead is a temporary result of a procurement pipeline, not yet persistent.

In other words, we try to procure. If it succeeds and the scheduler algorithm is happy with it, the spec is persisted and the task is launched. If not, this spec is thrown away.